moto/panorama/models.py

```diff
@@ -0,0 +1,304 @@
+            **{
+                k: v
+                for k, v in response_object.items()
+                if v is not None and k in static_response_fields
+            },
+            **{
+                "DeviceAggregatedStatus": self.device_aggregated_status,
+                "ProvisioningStatus": self.provisioning_status,
+            },
+        }
+
+    def response_listed(self) -> Dict[str, Any]:
+        response_object = super().gen_response_object()
+        response_object = deep_convert_datetime_to_isoformat(response_object)
+        static_response_fields = [
+            "Brand",
+            "CreatedTime",
+            "CurrentSoftware",
+            "Description",
+            "DeviceId",
+            "LastUpdatedTime",
+            "LatestDeviceJob",
+            "LeaseExpirationTime",
+            "Name",
+            "Tags",
+            "Type",
+        ]
+        return {
+            **{
+                k: v
+                for k, v in response_object.items()
+                if v is not None and k in static_response_fields
+            },
+            **{
+                "DeviceAggregatedStatus": self.device_aggregated_status,
+                "ProvisioningStatus": self.provisioning_status,
+            },
+        }
+
+    @property
+    def response_provision(self) -> Dict[str, Union[str, bytes]]:
+        return {
+            "Arn": self.arn,
+            "Certificates": base64.b64encode("certificate".encode("utf-8")).decode(
+                "utf-8"
+            ),
+            "DeviceId": self.device_id,
+            "IotThingName": self.iot_thing_name,
+            "Status": "AWAITING_PROVISIONING",
+        }
+
+    @property
+    def response_updated(self) -> Dict[str, str]:
+        return {"DeviceId": self.device_id}
+
+    @property
+    def response_deleted(self) -> Dict[str, str]:
+        return {"DeviceId": self.device_id}
+
+
+class PanoramaBackend(BaseBackend):
+    def __init__(self, region_name: str, account_id: str):
+        super().__init__(region_name, account_id)
+        self.devices_memory: Dict[str, Device] = {}
+
+    def provision_device(
+        self,
+        description: Optional[str],
+        name: str,
+        networking_configuration: Optional[Dict[str, Any]],
+        tags: Optional[Dict[str, str]],
+    ) -> Device:
+        device_obj = Device(
+            account_id=self.account_id,
+            region_name=self.region_name,
+            description=description,
+            name=name,
+            network_configuration=networking_configuration,
+            tags=tags,
+        )
+
+        self.devices_memory[device_obj.device_id] = device_obj
+        return device_obj
+
+    def describe_device(self, device_id: str) -> Device:
+        device = self.devices_memory.get(device_id)
+        if device is None:
+            raise ValidationError(f"Device {device_id} not found")
+        return device
+
+    @paginate(pagination_model=PAGINATION_MODEL)  # type: ignore[misc]
```

Comment:
Good to see this working OOTB! It's quite a lot of functionality that you get with just a single decorator, so I'm happy to see another example where this 'just' works.

Reply:
Yes, I'm also very impressed ! 💯 Moto is a very good package ;)